### PR TITLE
test: Use default 'background' in functional tests

### PR DESCRIPTION
**Pull request by justinmk**
Since deb18a050ef5/#9205 Nvim defaults to background=dark. But the
functional tests (`helpers.nvim_set`) specify background=light. Some tests
don't use `helpers.nvim_set`, which leads to a confusing inconsistency.

So remove background=light from `helpers.nvim_set`.

### CODE REVIEW

**Comment by justinmk** [comment]
After removing the `set background=light`, the test runner hangs. Adding `-c redraw!` un-hangs it:

```diff
diff --git a/test/functional/helpers.lua b/test/functional/helpers.lua
index ba00412423f6..fabd25a83e0b 100644
--- a/test/functional/helpers.lua
+++ b/test/functional/helpers.lua
@@ -38,7 +38,9 @@ local nvim_set  = 'set shortmess+=I noswapfile noautoindent'
                   ..' laststatus=1 undodir=. directory=. viewdir=. backupdir=.'
                   ..' belloff= noshowcmd noruler nomore'
 local nvim_argv = {nvim_prog, '-u', 'NONE', '-i', 'NONE', '-N',
-                   '--cmd', nvim_set, '--embed'}
+                   '--cmd', nvim_set,
+                   '-c', 'redraw!',
+                   '--embed'}
 -- Directory containing nvim.
 local nvim_dir = nvim_prog:gsub("[/\\][^/\\]+$", "")
 if nvim_dir == nvim_prog then
```

**Comment by bfredl** [comment]
If one attaches a debugger to nvim where it hangs, what is the traceback?

**Comment by justinmk** [comment]
Waiting on [reply from Travis](https://github.com/neovim/neovim/pull/9269) to enable debugging.

**Comment by bfredl** [comment]
Ah, so it happens only on travis, not locally?

**Comment by justinmk** [comment]
Yes, I can't reproduce it locally (nor on AppVeyor).

**Comment by justinmk** [comment]
The "hang" (which was consistent before) seems to have gone away. Reverting https://github.com/neovim/neovim/commit/f1ce9b3be2ea1e9e1dc5c6206bc441b310feb28b did not re-introduce the hang.

File: test/functional/ui/bufhl_spec.lua [hunk 1]
```diff
@@ -15,21 +15,25 @@ describe('Buffer highlighting', function()
     screen:attach()
     screen:set_default_attr_ids({
       [1] = {bold=true, foreground=Screen.colors.Blue},
-      [2] = {foreground = Screen.colors.Fuchsia}, -- String
-      [3] = {foreground = Screen.colors.Brown, bold = true}, -- Statement
-      [4] = {foreground = Screen.colors.SlateBlue}, -- Special
-      [5] = {bold = true, foreground = Screen.colors.SlateBlue},
-      [6] = {foreground = Screen.colors.DarkCyan}, -- Identifier
+      [2] = {foreground = tonumber('0xffa0a0')},                -- String/Constant
```
[the remark this answers is not in the document]
Comment (justinmk) [reply test/functional/ui/bufhl_spec.lua]: in `runtime/syntax/syncolor.vim`

**Comment by justinmk** [comment]
Obviated by https://github.com/neovim/neovim/pull/26334

**Pull request closed without merging**